arbor/backends/gpu/mechanism.cpp

```diff
@@ -156,7 +157,7 @@ void mechanism::instantiate(unsigned id,
     for (auto i: make_span(0, num_ions_)) {
         auto ion_binding = value_by_key(overrides.ion_rebind, ion_index_tbl[i].first).value_or(ion_index_tbl[i].first);
 
-        util::optional<ion_state&> oion = value_by_key(shared.ion_data, ion_binding);
+        ion_state* oion = value_by_key(shared.ion_data, ion_binding);
```

Comment:
Should this be `ptr_by_key`, not `value_by_key`?

Reply:
It should, yes.